ironfish/src/evm/evm.ts

```diff
@@ -96,6 +71,54 @@ export class IronfishEvm {
       await vm.evm.stateManager.revert()
     }
   }
+
+  decodeLogs(logs: Log[]): UTXOEvent[] {
+    const globalContract = new ethers.Interface(ContractArtifact.abi)
+
+    const events: UTXOEvent[] = []
+
+    for (const log of logs) {
+      // todo: placeholder until we determine an address
+      // if (Buffer.from(log[0]).toString('hex') !== 'globalContractAddress') {
+      //   continue
+      // }
+
+      try {
+        const [ironfishAddress, tokenId, caller, amount] = globalContract.decodeEventLog(
+          'Shield',
+          log[2],
+        )
+
+        events.push({
+          name: 'shield',
+          ironfishAddress: Buffer.from((ironfishAddress as string).slice(2), 'hex'),
+          caller: Address.fromString(caller as string),
+          assetId: this.getAssetId(caller as string, tokenId as bigint),
+          amount: amount as bigint,
+        })
+      } catch (e) {
+        try {
```

Comment:
Maybe separate try/catches and add a continue statement at end of each try?

Reply:
not sure what you mean here. if we `continue` then we just skip that log